test/transaction.test.js

```diff
@@ -0,0 +1,216 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: loopback-datasource-juggler
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+/* global getSchema:false */
+const DataSource = require('..').DataSource;
+const EventEmitter = require('events');
+const Connector = require('loopback-connector').Connector;
+const Transaction = require('loopback-connector').Transaction;
+const should = require('./init.js');
+
+let callCount;
+let transactionPassed;
+
+function resetState() {
```

Comment:
We prefer to keep helper functions (and class definitions) at the bottom of the test file and start with tests themselves - see http://loopback.io/doc/en/contrib/style-guide.html